issues building terraform-provider-ignition v2.1.2 with go version 1.18.5

We were having some touble building terraform-provider-ignition v2.1.2 https://github.com/community-terraform-providers/terraform-provider-ignition with go version 1.18.5.  It used to work fine with older go version i.e 1.15.10. 
However we had to upgrade the go version because of this PR - https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/436. This required us to upgrade to terraform >= 1.2.0 and go >= 1.18.1

This is the error we were seeing with go 1.18.5:
```
~/test4-11/go/src/github.com/community-terraform-providers/terraform-provider-ignition ~/test4-11/go/src/github.com/community-terraform-providers ~/test4-11/ocs-upi-kvm/scripts
==> Checking that code complies with gofmt requirements...
gofmt needs running on the following files:
./tools.go
You can use the command: `make fmt` to reformat code.
make: *** [GNUmakefile:35: fmtcheck] Error 1
``` 

As a workaround we have disabled the fmtcheck in GNUmakefile in our source code(https://github.com/ocp-power-automation/ocs-upi-kvm/blob/master/scripts/helper/create-cluster.sh#L339).

More details:
fmtcheck works fine with go 1.15.10:
```
[abhishek@nx142 terraform-provider-ignition]$ /test-go/bin/go version
go version go1.15.10 linux/ppc64le
[abhishek@nx142 terraform-provider-ignition]$ /test-go/bin/gofmt  -l `find . -name '*.go' | grep -v vendor`
[abhishek@nx142 terraform-provider-ignition]$
```
go 1.18.5 output:
```
[abhishek@nx142 terraform-provider-ignition]$ /test-go/bin/go version
go version go1.18.5 linux/ppc64le
[abhishek@nx142 terraform-provider-ignition]$ /test-go/bin/gofmt  -l `find . -name '*.go' | grep -v vendor`
./tools.go
```

--- a/tools.go
+++ b/tools.go
@@ -1,3 +1,4 @@
+//go:build tools
 // +build tools
 
 // tools is a dummy package that will be ignored for builds, but included for dependencies.

